open-craft/opencraft
# SE-1606: Periodic builds

This PR adds support for periodically building sandboxes.

**Dependencies**: None

**Testing instructions**:

1. deploy ocim (test on stage?)
1. enable periodic builds on an instance and set a reasonable interval for testing (~4h)
1. wait
1. verify that new appservers are being built and activated every `interval` time periods
1. probably more playing around to search for edge cases. (?)
1. view the settings table in the frontend for instances that do and don't have periodic builds enabled. Verify that the information displayed about periodic builds for said instances are correct.

**Screenshots**:

Frontend changes showing periodic builds status to user when disabled:

![1571026455](https://user-images.githubusercontent.com/9714796/66729162-23cfcb80-ee91-11e9-8652-73f4bd1d10dd.png)

enabled:

![1571026446](https://user-images.githubusercontent.com/9714796/66729164-24686200-ee91-11e9-808b-76a682ba1000.png)


**Author notes and concerns**:

- now is the time to agree on terminology. Please thoroughly check the names of the new table columns and help_text.

**Reviewers**
